app/rest/rest/src/main/java/io/syndesis/rest/metrics/prometheus/HttpQuery.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.rest.metrics.prometheus;
+
+import java.util.List;
+import javax.ws.rs.core.UriBuilder;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import org.immutables.value.Value;
+
+/**
+ * Prometheus HTTP Query
+ */
+@Value.Immutable
+@SuppressWarnings("immutables")
+public interface HttpQuery {
+
+    class Builder extends ImmutableHttpQuery.Builder {
+        // make ImmutableHttpQuery.Builder accessible
+    }
+
+    @Value.Immutable
+    @JsonDeserialize(builder = HttpQuery.LabelValue.Builder.class)
+    interface LabelValue {
+
+        @SuppressWarnings("PMD.UseUtilityClass")
+        class Builder extends ImmutableLabelValue.Builder {
+            public static HttpQuery.LabelValue of(final String value, final String label) {
+                return new Builder().label(label).value(value).build();
+            }
+        }
+
+        String getLabel();
+
+        String getValue();
+    }
+
+    String getHost();
+
+    String getMetric();
+
+    List<LabelValue> getLabelValues();
+
+    default UriBuilder getUriBuilder() {
+        StringBuilder queryExpression = new StringBuilder();
+        queryExpression.append(getMetric());
+        if (!getLabelValues().isEmpty()) {
+            queryExpression.append("%7B");
+            boolean first = true;
+            for (LabelValue label : getLabelValues()) {
+                if (first) {
+                    first = false;
+                } else {
+                    queryExpression.append(',');
+                }
+                queryExpression.append(label.getLabel());
+                queryExpression.append('=');
+                queryExpression.append('"');
+                queryExpression.append(label.getValue());
+                queryExpression.append('"');
```

Comment:
matter of tast, but I would delegate the adding of the value to the label itself (i.e. a `label.appendTo(queryExpression)` and with

```java
default void appendTo(StringBuilder params) {
    if (params.size() > 0) {
        params.add(",");
   }  
   params.append(getLabel()).append("=\"").append(getValue()).append('"');
}
```